Propagating changes in astromodels interface to HAL

henrikef opened the pull request:
This is needed to bring HAL up to date with the [dev branch](https://github.com/threeML/astromodels/tree/dev) of astromodels. Maybe don't merge it quite yet as it will break compatibility with the release versions and master branch of astromodels.

henrikef committed:
Propagating changes in astromodels interface to HAL

henrikef committed:
Merge branch 'master' into astromodels_interface

henrikef committed:
add test with dev branch

henrikef commented:
Now with added tests. Marked as draft to avoid accidental merge as long as the main branch is not ready.

henrikef committed:
Push to rerun tests

henrikef committed:
Merge branch 'master' into astromodels_interface

henrikef committed:
Fix macos version for tests

xiaojieww commented on .github/workflows/build_and_test.yml:
```diff
@@ -182,3 +182,85 @@ jobs:
+                  echo "Python version: ${{matrix.python-version}}"
+
+                  mamba install pytest codecov pytest-cov git astromodels threeML numba reproject root flake8
+                  pip install --no-binary :all: root_numpy
```
@henrikef  Should we change line 237 with `mamba install -c conda-forge root_numpy` as well? Since `pip install --no-binary :all: root_numpy` won't succeed while installation.

henrikef replied:
Sure, can you do it?

xiaojieww replied:
Sure, done!

xiaojieww committed:
Update build_and_test.yml

change `pip install --no-binary :all: root_numpy` to `mamba install -c conda-forge root_numpy`.

xiaojieww commented:
All Checks have passed, and I will Merge this pull request as we agreed. @henrikef
